src/components/compensations/CompensationSelector.tsx

```diff
@@ -27,18 +27,23 @@ export default function CompensationSelector({
 }: CompensationsProps) {
   const t = useTranslations("compensation_calculator");
 
+  const hasBenefits = (id: string) =>
+    compensations.benefitsByLocation.some((b) => b.location._ref === id);
+
+  const locationOptions: IOption[] = locations
+    .map((companyLocation) => ({
+      id: companyLocation._id,
+      label: companyLocation.companyLocationName,
+    }))
+    .filter((l) => hasBenefits(l.id));
+
   const [selectedLocation, setSelectedLocation] = useState<string>(
-    locations[0]._id,
+    locationOptions[0]?.id,
   );
   const [selectedLocationLabel, setSelectedLocationLabel] = useState<string>(
```

Comment:
We dont need this value if we filter the data. It's only used to get the correct city from translations.